test/integration/auth.js

```diff
@@ -154,6 +184,89 @@ function test(utils) {
       });
   }
 
+  function testListUsers() {
+    var promises = [];
+    uids.forEach(function(uid) {
+      var tempUserData = {
+        uid: uid,
+        password: 'password'
+      };
+      promises.push(admin.auth().createUser(tempUserData));
+    });
+    return Promise.all(promises)
+      .then(function() {
+        // Return 2 users with the provided page token.
+        // This test will fail if other users are created in between.
+        return admin.auth().listUsers(2, uids[0]);
+      })
+      .then(function(listUsersResult) {
+        // Confirm expected number of users.
+        utils.assert(
+          listUsersResult.users.length == 2,
```

Comment:
We could just test if the result has at most 2 users, and verify the types of the user attributes for each of them, instead of testing for exact matches. That will make this test case a bit more robust against changing project conditions.

Reply:
It's a fair point. I just feel more confident with test coverage on expected behavior with this check. If we get reports on its flakiness, feel free to modify it to not check for exact matches.